Sources/FTPropertyWrappers/Keychain - Support/KeychainCoding.swift

```diff
@@ -0,0 +1,371 @@
+            return storage
+        case MemoryLayout<UInt16>.size:
+            return UInt8(try decode(from: data) as UInt16)
+        case MemoryLayout<UInt32>.size:
+            return UInt8(try decode(from: data) as UInt32)
+        case MemoryLayout<UInt64>.size:
+            return UInt8(try decode(from: data) as UInt64)
+        default:
+            throw KeychainError.generalDecodingFailure
+        }
+    }
+
+    private func decode(from data: Data) throws -> UInt16 {
+        switch data.count {
+        case MemoryLayout<UInt8>.size:
+            return UInt16(try decode(from: data) as UInt8)
+        case MemoryLayout<UInt16>.size:
+            let storage = data.withUnsafeBytes { $0.load(as: UInt16.self) }
+            return storage
+        case MemoryLayout<UInt32>.size:
+            return UInt16(try decode(from: data) as UInt32)
+        case MemoryLayout<UInt64>.size:
+            return UInt16(try decode(from: data) as UInt64)
+        default:
+            throw KeychainError.generalDecodingFailure
+        }
+    }
+
+    private func decode(from data: Data) throws -> UInt32 {
+        switch data.count {
+        case MemoryLayout<UInt8>.size:
+            return UInt32(try decode(from: data) as UInt8)
+        case MemoryLayout<UInt16>.size:
+            return UInt32(try decode(from: data) as UInt16)
+        case MemoryLayout<UInt32>.size:
+            let storage = data.withUnsafeBytes { $0.load(as: UInt32.self) }
+            return storage
+        case MemoryLayout<UInt64>.size:
+            return UInt32(try decode(from: data) as UInt64)
+        default:
+            throw KeychainError.generalDecodingFailure
+        }
+    }
+
+    private func decode(from data: Data) throws -> UInt64 {
+        switch data.count {
+        case MemoryLayout<UInt8>.size:
+            return UInt64(try decode(from: data) as UInt8)
+        case MemoryLayout<UInt16>.size:
+            return UInt64(try decode(from: data) as UInt16)
+        case MemoryLayout<UInt32>.size:
+            return UInt64(try decode(from: data) as UInt32)
+        case MemoryLayout<UInt64>.size:
+            let storage = data.withUnsafeBytes { $0.load(as: UInt64.self) }
+            return storage
+        default:
+            throw KeychainError.generalDecodingFailure
+        }
+    }
+
+    private func decode(from data: Data) throws -> Float {
+        fatalError("Encoding root type Float not supported!")
+    }
+
+    private func decode(from data: Data) throws -> Float80 {
+        fatalError("Encoding root type Float80 not supported!")
+    }
+
+    private func decode(from data: Data) throws -> Double {
+        fatalError("Encoding root type Double not supported!")
+    }
+
+    private func decode(from data: Data) throws -> Bool {
+        let value: Int = try decode(from: data)
+        switch value {
+        case 0:
+            return false
+        case 1:
+            return true
```

Comment:
What if we used C-style booleans? Zero is true and anything else is false?

Reply:
Treating booleans as a number is considered a bad thing even amongst C developers (I've met). Boolean is a logical value, not a number. We encode "true" as 1, not as "any positive number". Since we know, that bool is encoded by this encoder every single time as 0 or 1 (or any other pair of constants) it makes sense to indicated other values as errors - which they are.

Reply:
That goes against thing you said earlier, that the property encoder should allow simply getting values from previous versions of apps. This opens another question I forgot to ask: Do we plan to support custom encoding/decoding?